contrib/guix/guix-build

```diff
@@ -146,9 +146,12 @@ avail_KiB="$(df -Pk "$VERSION_BASE" | sed 1d | tr -s ' ' | cut -d' ' -f4)"
 total_required_KiB=0
 for host in $HOSTS; do
     case "$host" in
-        *darwin*) required_KiB=440000 ;;
-        *mingw*)  required_KiB=7600000 ;;
-        *)        required_KiB=6400000 ;;
+        *darwin*)       required_KiB=440000 ;;
+        *mingw*)        required_KiB=8700000 ;;
+        *x86_64-linux*) required_KiB=6100000 ;;
+        *arm-linux*)    required_KiB=4800000 ;;
+        *riscv64*)      required_KiB=6600000 ;;
+        *)              required_KiB=7400000 ;;
```

Comment:
why is this 7400000? this line is for matching any host, correct? shouldn't it be the max required, then (8700000, in this case)

Reply:
> this line is for matching any host, correct?

For matching any _unspecified above_ host.

Reply:
gotcha. wouldn't it still be better to make this the same as the max value of the specified hosts? the system doing the building can already handle that value because we are specifying it for a known host, so why not play it safe for an unspecified host?

i guess i don't understand how this value was arrived at or why we would make it less than the max

Reply:
The default value `required_KiB=7400000` is specified for the sake of the code brevity.
Actually it covers the following hosts:
- aarch64-linux-gnu -- 7266380 KiB
- powerpc64-linux-gnu -- 7330884 KiB
- powerpc64le-linux-gnu -- 7293340 KiB

Overestimating of the required free disk space seems suboptimal in cases when Guix builds are being done in virtual machines with limited resources.

Alternatively, each host can be specified explicitly.

Reply:
Thank you for the explanation. I'm not sure I totally agree and would prefer that if the HOST is an unspecified HOST, we set `total_required_KiB` to `avail_KiB`. 

I think it would be far worse to fail during a long-running process due to underestimating the space required vs finishing the build but with more space than was needed.

Said differently, if we haven't specified anything for a host, assume the worst and use all the space available.

Reply:
> Said differently, if we haven't specified anything for a host, assume the worst and use all the space available.

The `HOSTS` variable is always defined here: https://github.com/bitcoin/bitcoin/blob/aa9a48bb8af7a7f1bef98051a1590de3713c7789/contrib/guix/guix-build#L76-L79

Reply:
right, but I can also run `HOSTS=foo-bar-baz ./contrib/guix-build`. In this case, I think it would be better for the default to be `required_KiB = avail_KiB`

Reply:
AH, my apologies. I misunderstood your explanation. The last case is to cover:

* aarch64-linux-gnu -- 7266380 KiB
* powerpc64-linux-gnu -- 7330884 KiB
* powerpc64le-linux-gnu -- 7293340 KiB

in that case, i would recommend `*-linux-gnu` instead of `*`. this makes it more readable/explicit as to what the code is doing and which subgroup from `HOSTS` is being matched.